.github/workflows/on-merge-queue.yml

```diff
@@ -0,0 +1,65 @@
+---
+# Merge queue.  A PR that reaches here has already passed the more rigorous PR
+# suite, so is very likely to pass.  The main reasons for failures here are
+# flaky VMs timing out (which we can't do much about), or a merge conflict
+# with another PR that is also in the merge queue.
+#
+# There's no reason to have multiple stages in this case, because we're
+# expecting it to pass.  Having more than one stage frustrates parallel
+# throughput in low-contention cases, and guarantees a longer critical path.
+name: Merge queue
+on:
+  merge_group:
+    branches: [main, "stable/*"]
```

Comment:
I'm wondering what's going to happen after we merge this. We probably aren't going to backport this PR and we'll still use azure on stable branches until they age out (`stable/2.0` goes EoL in about a month). We might want to backport it to `stable/1.4` since that's supported longer. I'm wonder if we will need to remove this until we are using gha on all the supported stable branches.